pkg/generators/readers.go

```diff
@@ -237,6 +238,76 @@ func (g *ReadersGenerator) generateHelpers() error {
 			}
 			return
 		}
+
+		// ParseInteger reads a string and parses it to integer,
+		// if an error ocurred it returns a non-nil error.
+		func ParseInteger(query url.Values, parameterName string) (*int, error) {
+			s := query.Get(parameterName)
+			if s == "" {
```

Comment:
This will return no error when the user sends a request with the parameter and no value. That isn't correct. Only if the parameter doesn't have a value should we return `nil`. It doesn't check if there are more than one value, which isn't correct either. I'd suggest something like this:

```go
func ParseInteger(quer url.Values, name string) (result *int, err error) {
        values := query[parameterName]
        count := len(values)
        if count == 0 {
                return
        }
       if size > 1 {
               err = fmt.Errorf(
                       "expected at most one value for parameter '%s' but got %d",
                       name, size,
               )
               return
        }
        value := values[0]
        result, err = strconv.ParseInt(value, 10, 64)
        if err != nil {
                err = fmt.Errorf(
                        "value '%s' isn't a valid for the '%s' parameter because it isn't an integer",
                        value, name,
                 )
                 return
        }
        return
}
```

Same for the other types.

Remember the import the packages needed, like `net/url`.